core/templates/dev/head/domain/exploration/WrittenTranslationsObjectFactorySpec.ts

```diff
@@ -0,0 +1,189 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for WrittenTranslationsObjectFactory.ts
+ */
+import { TestBed } from '@angular/core/testing';
+
+import { WrittenTranslationsObjectFactory } from
+  'domain/exploration/WrittenTranslationsObjectFactory';
+import { WrittenTranslationObjectFactory } from
+  'domain/exploration/WrittenTranslationObjectFactory';
+
+describe('Written Translations Object Factory', () => {
+  let writtenTranslationsObjectFactory;
+  let writtenTranslationObjectFactory;
+  let writtenTranslationsBackendDict;
+
+  beforeEach(() => {
+    writtenTranslationsObjectFactory = TestBed.get(
+      WrittenTranslationsObjectFactory);
+    writtenTranslationObjectFactory = TestBed.get(
+      WrittenTranslationObjectFactory);
+
+    writtenTranslationsBackendDict = (
+      writtenTranslationsObjectFactory.createFromBackendDict({
+        translations_mapping: {
+          content_1: {
+            'hi-en': {
+              html: '',
+              needs_update: false
+            }
+          }
+        }
+      }));
+  });
+
+  it('should create a written translations object from backend', () => {
+    expect(writtenTranslationsBackendDict.toBackendDict())
+      .toEqual({
+        translations_mapping: {
+          content_1: {
+            'hi-en': {
+              html: '',
+              needs_update: false
+            }
+          }
+        }
+      });
+  });
+
+  it('should create an empty written translations object', () => {
+    const emptyWrittenTranslationsObject = (
+      writtenTranslationsObjectFactory.createEmpty());
+    expect(emptyWrittenTranslationsObject.getAllContentId()).toEqual([]);
+  });
+
+  it('should add and delete contents from a written translations object',
+    () => {
+      writtenTranslationsBackendDict.addContentId('content_2');
+      expect(() => {
+        writtenTranslationsBackendDict.addContentId('content_2');
+      }).toThrow(Error('Trying to add duplicate content id.'));
+      expect(writtenTranslationsBackendDict.getAllContentId()).toEqual([
+        'content_1', 'content_2']);
+
+      writtenTranslationsBackendDict.deleteContentId('content_2');
+      expect(writtenTranslationsBackendDict.getAllContentId()).toEqual([
```

Comment:
Check the contents of the content id array after each operation you do.

Reply:
Done.